test/measurement/discriminator/test_iq_discriminator.py

```diff
@@ -148,3 +148,28 @@ def filter_and_discriminate(self):
 
         self.assertEqual(discriminator.discriminate([[i0, q0]])[0], '0')
         self.assertEqual(discriminator.discriminate([[i1, q1]])[0], '1')
+
+    def test_plot(self):
+        """
+        Unit test to check the plotting routine.
+        """
+        discriminator = LinearIQDiscriminator(self.cal_results, self.qubits,
+                                              ['0', '1'])
+
+        ax, fig = discriminator.plot()
+
+        fig.show()
```

Comment:
You don't want to do this in a test. Depending on the matplotlib backend and system configuration this could cause an error. For automated testing you need to ensure that if you output an image you write it to disk with `savefig()`. That being said there isn't really a reason to display the output image here, it's not used as part of the test and you're inspecting the resulting mpl objects to verify they're somewhat sane.

Reply:
Thanks for catching this. I removed the `fig.show()`.